packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -77,79 +76,80 @@ class SharedPreferences {
   ///
   /// It is NOT guaranteed that this cache and the device prefs will remain
   /// in sync since the setter method might fail for any reason.
-  final Map<String, Object> _preferenceCache;
+  final Map<String, Object?> _preferenceCache;
```

Comment:
Why are the values nullable? That's not what the platform interface that this is caching vends.